### ISSUE TITLE
Unnecessary return value of `BaseStorage.set_trail_intermediate_value`.

### ISSUE THREAD

**Issue by ytsmiling** [optuna/optuna#1309]
## TL;DR
This PR requests to deprecate the return value of [`BaseStorage.set_trial_intermediate_value`](https://github.com/optuna/optuna/blob/24300dc57661bff7961499a2d52534fcd41ad258/optuna/storages/base.py#L144) method.

## General background
[`BaseStorage.set_trial_intermediate_value`](https://github.com/optuna/optuna/blob/24300dc57661bff7961499a2d52534fcd41ad258/optuna/storages/base.py#L144) method returns False when the specified step already has a value. Its motivation is identical with #1308. The method is solely used in [`trial.report`](https://github.com/optuna/optuna/blob/v1.4.0/optuna/trial.py#L807), but the return value is ignored.

## Proposal
Remove the return value and change the method's behavior to "always overwrite existing values". See #1308 for similar discussion.

## Note
This is out of the scope of this issue, but #852 is a related issue and we should add appropriate handling in [`trial.report`](https://github.com/optuna/optuna/blob/v1.4.0/optuna/trial.py#L807).
